### ISSUE TITLE
Implement `onWorkerEnd` hook

### ISSUE THREAD

**Issue by suchitrak** [webdriverio/webdriverio#6449]
I am trying to generate two reports `ie_feature_name_report.json `and `firefox_feature_name_report.json` for a feature file that is been executed in parallel on two browsers. But it is generating just one report `feature_name.json`

I have tried following ways. But none of them works as expected. 

wdio config 

```
capabilities: [
    {
      browserName: 'firefox',
    },
    {
      browserName: 'internet explorer',
    },
  ],
onWorkerStart: function (cid, caps, specs, args, execArgv) {
        console.log(caps)
        folder_name = caps.browserName
        console.log(folder_name)
    },
reporters: [
        [
            'cucumberjs-json',
            {
                jsonFolder: `webdriverio-test-reports1/${folder_name}`,
                language: 'en',
            },
        ],
]
```

The report with name `<feature_file_name>.json` is created.   As we cannot format the file name of the report. The report of parallel execution gets overwritten.

If there was a hook `"onWorkerEnd".` I could have copied the <feature_file_name>.json  report to a new folder. I did not find a webdriverio hook, that would be executed at the end of a parallel execution( I mean when tests on firefox completes execution). Is there a webdriverio hooks that could be executed at start and end of each parallel browser execution.

I also tried "`wdio-multiple-cucumber-html-reporter`" but that creates two identical html reports which is of no use.

**Comment by christian-bromann** [comment]
Thanks for filing the issue. I recommend to raise this problem in the [reporter project](https://github.com/wswebcreation/wdio-cucumberjs-json-reporter) as it should provide the ability to run tests in parallel. If this is about creating a `onWorkerEnd` hook it would be fine here.

**Comment by suchitrak** [comment]
I have created the issue in https://github.com/wswebcreation/wdio-cucumberjs-json-reporter/issues/46 . But it would also be nice to have `onWorkerEnd` . For parallel executions if we want to lock/unlock the user that was been used, it would be good to have this hook.

Say I define
`accounts.config.js` 

```
  stagingEnv: {
        admin: [{
            login: 'admin_user_1@abcd.com',
            password: 'test1234',
            flag: unlocked
        },
        {
            login: 'admin_user_2@abcd.com',
            password: 'test1234',
            flag: unlocked
        }
        {
            login: 'admin_user_3@abcd.com',
            password: 'test1234',
            flag: unlocked
        }],
        user: [{
            login: 'user_1_@abcd.com',
            password: 'test1234',
            flag: unlocked
        },
        {
            login: 'user_2_@abcd.com',
            password: 'test1234',
            flag: unlocked
        },
        {
            login: 'user_3_@abcd.com',
            password: 'test1234',
            flag: unlocked
        },]
```

wdio.conf.js

```
onWorkerStart: function (cid, caps, specs, args, execArgv) {
        if(users[args.env][index] == "unlocked")
        userName = users[args.env][index].login
        users[args.env][index].flag = "locked"
    },

onWorkerEnd: function (cid, caps, specs, args, execArgv) {
        users[args.env][index].flag = "unlocked"
    },
```

This was we could manage the user-logins been used for parallel execution

**Comment by christian-bromann** [comment]
Alright, sounds good to me. If anyone interested, feel free to pick this up ☺️ I am happy to support.

**Comment by christian-bromann** [comment]
Duplicate of #6356
